server/bin/pbench-audit-server.py

```diff
@@ -0,0 +1,445 @@
+        u = os.path.basename(user)
+        if os.path.isdir(user):
+            user_dir.append(user)
+        else:
+            unexpected_objects.append(u)
+
+    if len(unexpected_objects)>0:
+        output_format(unexpected_objects, "Unexpected files found:")
+        cnt = cnt+1
+
+    if len(user_dir)>0:
+        for user in user_dir:
+            cnt = verify_controllers(user,config)
+            if cnt>0:
+                cnt = cnt+1
+    
+    return cnt
+
+def verify_prefixes(controller, cnt):
+    controller_arg = os.path.join(controller,".prefix")
+    if not os.path.exists(controller_arg):
+        return 0
+    if not os.path.isdir(controller_arg):
+        print("\t* Prefix directory, .prefix, is not a directory!")
+        return 1
+    cnt=0
+
+    non_prefixes, wrong_prefixes = ([] for i in range(2))
+    controllers = glob.iglob(os.path.join(controller_arg,"*"))
+
+    for dirent in controllers:
+        d = os.path.basename(dirent)
+        if not d.startswith('prefix.') and not d.endswith('.prefix'):
+            non_prefixes.append(d)
+        elif d.startswith('prefix.'):
+            wrong_prefixes.append(d)
+
+    if cnt==0 and (len(non_prefixes)>0 or len(wrong_prefixes)>0):
+        print("\nController: ", os.path.basename(controller))
+
+    if len(non_prefixes)==0 and len(wrong_prefixes)==0:
+        cnt=cnt+1
+
+    if len(non_prefixes)>0:
+        output_format(non_prefixes, "Unexpected file system objects in .prefix directory:")
+        cnt = cnt+1
+    if len(wrong_prefixes)>0:
+        output_format(wrong_prefixes, "Wrong prefix file names found in /.prefix directory:")
+        cnt = cnt+1
+
+    return cnt
+
+def output_format(n, msg):
+    print("\t*",msg)
+    print("\t  ++++++++++", end='\n\t\t  ')
+    print(*n, sep='\n\t\t  ')
+    print("\t  ----------")
+
+def verify_archive(config):
+
+    archive = config.ARCHIVE
+    cnt=0
+    controllers = glob.iglob(os.path.join(archive,'*'))
+    bad_controllers=[]
+
+    for controller in controllers:
+        if os.path.isdir(controller):
+            cnt = verify_subdirs(config, controller)
+            cnt = verify_prefixes(controller, cnt)
+        else:
+            bad_controllers.append(os.path.basename(controller))
+    
+    if len(bad_controllers)>0:
+        print("\nController: ", os.path.basename(controller))
+        output_format(bad_controllers, "Bad-Controllers")
+        cnt = cnt+1
+
+    return cnt
+
+def main():
+    cfg_name = os.environ.get("_PBENCH_SERVER_CONFIG")
+    if not cfg_name:
+        print("{}: ERROR: No config file specified; set CONFIG env variable or"
+                " use --config <file> on the command line".format(_NAME_),
+                file=sys.stderr)
+        return 2
+
+    try:
+        config = PbenchConfig(cfg_name)
+    except BadConfig as e:
+        print("{}: {}".format(_NAME_, e), file=sys.stderr)
+        return 1
+
+    logger = get_pbench_logger(_NAME_, config)
+    archive = config.ARCHIVE
```

Comment:
What benefit does this provide?  It seems like it is used with `print` statements below where it doesn't apply, and we pass `config` to the various functions.